### PR TITLE
Borg, Gripper, and Integrated Circuit interaction update.

### PR DESCRIPTION
**Pull request by Mechoid**
From the changelog because I really don't want to retype all of this manually:
changes: 
  - add: "Research and Engineering borgs now have 'Circuit Grippers', used for constructing and operating integrated circuit machinery."
  - add: "Grippers now allow interaction with their contents, by attacking the gripper itself with an item when one is held. I.E., drawing from a beaker with a syringe."
  - tweak: "Grippers now show their held item's examine information when examined. Tab information added."
  - tweak: "Cyborgs can now interact with integrated circuit printers and machines when adjacent to them."
  - tweak: "Multitools now have a menu to switch modes between standard use and integrated circuit reference scanning. Antag multi-tools untouched for now."
  - bugfix: "Integrated circuit printer now respects adjacency, if it is not set to debug."

### CODE REVIEW

**Comment by Anewbe** [comment]
What exactly does the multitool part of this have to do with any of the rest of it?

**Comment by Mechoid** [comment]
Integrated circuitry usage, so it has full functionality of the debugger and wirer.

File: code/game/objects/items/devices/multitool.dm [hunk 1]
```diff
@@ -18,20 +18,54 @@
...
+				accepting_refs = 1
+		if("Switch Mode")
+			mode_switch(user)
+
+	update_icon()
+
+	return ..()
+
+/obj/item/device/multitool/proc/mode_switch(mob/living/user)
+	if(mode_index >= modes.len)
```
Comment (Atermonera) [inline code/game/objects/items/devices/multitool.dm]: `if(++mode_index > modes.len) mode_index = 1` will exclude the else statement
Or `++mode_index %= modes.len`, but that's maybe not as clear

File: code/game/objects/items/devices/multitool.dm [hunk 2]
```diff
@@ -18,20 +18,54 @@
...
+		mode_index = 1
 	else
-		..()
+		mode_index++
+
+	toolmode = modes[mode_index]
+	to_chat(user,"<span class='notice'>\The [src] is now set to [toolmode].</span>")
+
+	if(toolmode != MULTITOOL_MODE_INTCIRCUITS)
```
Comment (Atermonera) [inline code/game/objects/items/devices/multitool.dm]: `accepting_refs = (toolmode == MULTITOOL_MODE_INTCIRCUITS)` will eliminate the whole if-else block

**Pull request merged**